backend/src/permissions.js

```diff
@@ -0,0 +1,18 @@
+import { allow, rule, shield } from "graphql-shield";
+
+const isAuthenticated = rule({ cache: "contextual" })(async (parent, args, ctx, info) => {
+    return ctx.userId !== null;
+});
+
+export const permissions = shield(
+    {
+        Mutation: {
+            signup: allow,
+            login: allow,
+        },
+    },
+    {
+        allowExternalErrors: true,
+        fallbackRule: isAuthenticated,
```

Comment:
Could be a bit more secure with a `fallbackRule` of `deny` - but then you'll have to add a lot more rules explicitly.
OK for me